corehq/apps/users/views/mobile/users.py

```diff
@@ -679,6 +685,226 @@ def post(self, request, *args, **kwargs):
+            'editUrl': reverse(EditCommCareUserView.urlname, args=[self.domain, user.user_id]),
+            'deactivateUrl': "#",
+            'actionText': _("Deactivate") if user.is_active else _("Activate"),
+            'action': 'deactivate' if user.is_active else 'activate',
+        }
+
+    def _user_query(self, search_string, page, limit):
+        default_search_fields = ["username", "last_name", "first_name"]
+        return (UserES()
+                .domain(self.domain)
+                .mobile_users()
+                .search_string_query(search_string, default_search_fields)
+                .start(limit * (page - 1))
+                .size(limit))
+
+    @allow_remote_invocation
+    def get_pagination_data(self, in_data):
+        if not isinstance(in_data, dict):
+            return {
+                'success': False,
+                'error': _("Please provide pagination info."),
+            }
+        try:
+            limit = int(in_data.get('limit', 10))
+        except ValueError:
+            limit = 10
+
+        page = in_data.get('page', 1)
+        query = in_data.get('query')
+
+        users_query = self._user_query(query, page, limit)
+        if in_data.get('showDeactivatedUsers', False):
+            users_query = users_query.show_only_inactive()
+        users_data = users_query.run()
+        return {
+            'response': {
+                'itemList': map(self._format_user, users_data.hits),
+                'total': users_data.total,
+                'page': page,
+                'query': query,
+            },
+            'success': True,
+        }
+
+    @allow_remote_invocation
+    def modify_user_status(self, in_data):
+        if not self.can_add_extra_users:
+            return {
+                'error': _("No Permission."),
+            }
+        try:
+            user_id = in_data['user_id']
+        except KeyError:
+            return {
+                'error': _("Please provide a user_id."),
+            }
+        try:
+            is_active = in_data['is_active']
+        except KeyError:
+            return {
+                'error': _("Please provide an is_active status."),
+            }
+        user = CommCareUser.get_by_user_id(user_id, self.domain)
+        user.is_active = is_active
+        user.save()
+        return {
+            'success': True,
+        }
+
+    @allow_remote_invocation
+    def check_username(self, in_data):
+        try:
+            username = in_data['username'].strip()
+        except KeyError:
+            return HttpResponseBadRequest('You must specify a username')
+        if username == 'admin' or username == 'demo_user':
+            return {'error': _('Username {} is reserved.'.format(username))}
+        if '@' in username:
+            return {
+                'error': _('Username {} cannot contain "@".'.format(username))
+            }
+        if ' ' in username:
```

Comment:
Should this check for any type of whitespace?  It might be more comprehensive to say "usernames can only contain letters and numbers"